ds701_book/14-Classification-I-Decision-Trees.qmd

```diff
@@ -1,407 +1,1158 @@
+    node [shape=oval, style="filled", color="black", fontname="helvetica"] ;
+    edge [fontname="helvetica"] ;
+    0 [label="Home\\nOwner", fillcolor="#ffffff"] ;
+    1 [label="Yes: 0\\nNo: 3", fillcolor="#ffffff", shape="square"] ;
+    2 [label="Yes: 3\\nNo: 4", fillcolor="#ffffff", shape="square"] ;
+    0 -> 1 [labeldistance=2.5, labelangle=45, headlabel="Yes"] ;
+    0 -> 2 [labeldistance=2.5, labelangle=-45, headlabel="No"] ; }
+"""
 
-In a __Binary split__ we divide the values into two groups.  
+# Use graphviz to render the dot file
+graph = graphviz.Source(dot_data)  
+graph
+```
 
-In this case, we need to find an optimal partitioning of values into groups.
+``` {python}
+#| echo: false
+#| fig-cap: '(b) Collective Entropy: 0.686'
+dot_data = """
+digraph Tree {
+    node [shape=oval, style="filled", color="black", fontname="helvetica"] ;
+    edge [fontname="helvetica"] ;
+    0 [label="Marital\\nStatus", fillcolor="#ffffff"] ;
+    1 [label="Yes: 2\\nNo: 3", fillcolor="#ffffff", shape="square"] ;
+    2 [label="Yes: 0\\nNo: 3", fillcolor="#ffffff", shape="square"] ;
+    3 [label="Yes: 1\\nNo: 1", fillcolor="#ffffff", shape="square"] ;
+    0 -> 1 [xlabel="Single"] ;
+    0 -> 2 [label="Married"] ;
+    0 -> 3 [xlabel="Divorced"] ; }
+"""
+
+# Use graphviz to render the dot file
+graph = graphviz.Source(dot_data)  
+graph
+```
 
-![](figs/L14-split-2.png){width="600px"}
+``` {python}
+#| echo: false
+#| fig-cap: '(c) Collective Entropy index: 0.00'
+dot_data = """
+digraph Tree {
+    node [shape=oval, style="filled", color="black", fontname="helvetica"] ;
+    edge [fontname="helvetica"] ;
+    0 [label="ID", fillcolor="#ffffff"] ;
+    1 [label="Yes: 0\\nNo: 1", fillcolor="#ffffff", shape="square"] ;
+    2 [label="...", fillcolor="#ffffff", shape="none"] ;
+    3 [label="Yes: 1\\nNo: 0", fillcolor="#ffffff", shape="square"] ;
+    4 [label="Yes: 0\\nNo: 1", fillcolor="#ffffff", shape="square"] ;
+    5 [label="Yes: 1\\nNo: 0", fillcolor="#ffffff", shape="square"] ;
+    0 -> 1 [xlabel="1"] ;
+    0 -> 2 [xlabel="2", color="white"] ;
+    0 -> 3 [xlabel="8"] ;
+    0 -> 4 [xlabel="9"] ;
+    0 -> 5 [xlabel="10"] ; }
+"""
+
+# Use graphviz to render the dot file
+graph = graphviz.Source(dot_data)  
+graph
+```
 
----
+:::
+
+::: {.callout-tip}
+Try calculating the collective Entropy for (a) and (b) and see if you get the same values.
+:::
+
+::: {.callout-important}
+The collective entropy for (c) is 0. Why would we not want to use this node?
+:::
+
+## Gain Ratio
 
-An __Ordinal__ attribute is similar to a Nominal attribute, except that since
-attributes have an ordering, we can specify the test in terms of a threshold.  
+From [@Tan2018, p. 127]:
 
-This simplifies the search for a good paritition.
+* Having a low impurity value alone is insufficient to find a good attribute test condition for a node. 
+* Having more number of child nodes can make a decision tree more complex and consequently more susceptible to overfitting. 
```

Comment:
I think you were trying to say either:

Having a larger number of child nodes...

OR

Having more child nodes ...